apps/web/app/components/ui/LanguageEditor/LanguageEditor.vue

```diff
@@ -1,47 +1,73 @@
 <template>
-  <SfTooltip :label="languageLabel" placement="bottom" :show-arrow="true">
-    <div v-if="!loading" class="flex items-center space-x-1 md:space-x-2">
-      <SfIconLanguage class="w-4 h-4 md:w-6 md:h-6" />
-      <div class="relative flex items-center">
-        <select
-          :value="currentLocale"
-          class="form-select focus:outline-none focus:ring-0 focus:border-transparent text-sm md:text-base appearance-none cursor-pointer"
-          data-testid="editor-language-select"
-          @input="(ev: any) => switchLanguage(ev.target.value)"
-        >
-          <option
-            v-for="locale in availableLocales"
-            :key="locale"
-            :value="locale"
-            class="font-medium text-sm md:text-base"
-          >
-            {{ t(`lang.${locale}`) }}
-          </option>
-        </select>
-        <div class="flex items-center">
-          <SfIconExpandMore class="w-4 h-4 md:w-6 md:h-6" />
-        </div>
-      </div>
-    </div>
-  </SfTooltip>
+  <div v-if="!loading" ref="root" class="relative flex items-center space-x-1 md:space-x-2">
+    <SfTooltip :label="languageLabel" placement="left" :show-arrow="true">
+      <button
+        type="button"
+        class="flex items-center focus:outline-none"
+        :aria-expanded="isOpen"
+        aria-haspopup="listbox"
+        @click="toggle"
+      >
+        <SfIconLanguage class="w-4 h-4 md:w-6 md:h-6" />
+        <span class="ml-1 text-sm md:text-base">
+          {{ currentLabel }}
+        </span>
+        <SfIconExpandMore class="w-4 h-4 md:w-6 md:h-6" />
+      </button>
+    </SfTooltip>
+
+    <UiLanguageList
+      :is-open="isOpen"
+      :active-index="activeIndex"
+      @locale-selected="handleLocaleSelected"
+      @active-index-change="activeIndex = $event"
+    />
+  </div>
 </template>
 
 <script setup lang="ts">
 import { SfIconLanguage, SfIconExpandMore, SfTooltip } from '@storefront-ui/vue';
-import type { Locale } from 'vue-i18n';
 
 const languageLabel = 'Change the active language to manage multilingual content.';
 
-const { locale: currentLocale, availableLocales } = useI18n();
-const { switchLocale } = useLocalization();
-const switchLanguage = async (locale: Locale) => {
-  await switchLocale(locale, false);
-};
+const { locale: currentLocale, availableLocales, t } = useI18n();
```

Comment:
dont use `t` from useI18n(). you can use `t` anywhere because we override it in our shop-core module.
```suggestion
const { locale: currentLocale, availableLocales } = useI18n();
```